stanford-sw/src/edu/stanford/StanfordIndexer.java

```diff
@@ -588,6 +588,64 @@ MSS MEDIA (e.g. MSS MEDIA 0025), MSS PHOTO (e.g. MSS PHOTO 0463), MSS PRINTS (e.
     return physicalFormats;
   }
 
+  public Set<String> getAllGenres(final Record record)
+  {
+    Set<String> freeFormGenres = new HashSet<String>();
+    Set<String> genres = new HashSet<String>();
+    Set<String> allGenres = new HashSet<String>();
+    freeFormGenres = getFreeFormGenre(record, "[\\\\,;:]", "([\\p{L}\\p{N}]{4}|\\.*?[\\s)]|[..{2,}]|[AMUaw][adir][cirt])");
+    genres = getGenres(record);
+    allGenres.addAll(freeFormGenres);
+    allGenres.addAll(genres);
+
+    return allGenres;
+  }
+
+  /**
+   * Returns all 655av and the subfield v (repeatable) in list of 6xx fields
+   * @param record a marc4j Record object
+     * @param charsToReplaceRegEx a regular expression of trailing chars to be
+     *   replaced (see java Pattern class).  Note that the regular expression
+     *   should NOT have '$' at the end.
+     *   (e.g. " *[,/;:]" replaces any commas, slashes, semicolons or colons
+     *     at the end of the string, and these chars may optionally be preceded
+     *     by a space)
+     * @param charsB4periodRegEx a regular expression that must immediately
+     *  precede a trailing period IN ORDER FOR THE PERIOD TO BE REMOVED.
+     *  Note that the regular expression will NOT have the period or '$' at
+     *  the end.
+     *   (e.g. "[a-zA-Z]{3,}" means at least three letters must immediately
+     *   precede the period for it to be removed.)
+   *
+   * @return Set of strings containing genre_ssim values without trailing chars
+   */
+    @SuppressWarnings("unchecked")
+  public Set<String> getFreeFormGenre(final Record record, String charsToReplaceRegEx, String charsB4periodRegEx)
```

Comment:
since you're only calling this internally once, I would move the regex into the method (sort of like "desiredTagFldSpec") and take them out of the params list.  Then it simplifies the method call above, too.